File: api/store1/src/main/java/com/mrcl/store1/admin/dto/AdminProductRow.java
package com.mrcl.store1.admin.dto;

import java.math.BigDecimal;
import java.util.Date;

/**
 * DTO returned in admin product listing.
 */
public record AdminProductRow(
        Long id,
        Long categoryId,
        String categoryName,
        String sku,
        String name,
        String description,
        BigDecimal unitPrice,
        String imageUrl,
        boolean active,
        int unitsInStock,
        Date dateCreated,
        Date lastUpdated
) {}

File: ui-frontent/loja/src/app/common/admin/admin-product-row.ts
export interface AdminProductRow {
  id: number;
  categoryId: number;
  categoryName: string;
  sku: string;
  name: string;
  description: string;
  unitPrice: number;
  imageUrl: string;
  active: boolean;
  unitsInStock: number;
  dateCreated: string;
  lastUpdated: string | null;
}